cli/flox/src/commands/generations/mod.rs

```diff
@@ -0,0 +1,58 @@
+use anyhow::{Result, bail};
+use bpaf::Bpaf;
+use flox_rust_sdk::flox::Flox;
+use flox_rust_sdk::models::environment::ConcreteEnvironment;
+use flox_rust_sdk::models::environment::generations::{
+    AllGenerationsMetadata,
+    GenerationsEnvironment,
+};
+use indoc::formatdoc;
+use list::List;
+use tracing::instrument;
+
+use super::display_help;
+use crate::commands::environment_description;
+use crate::config::Config;
+
+mod list;
+
+/// Generations Commands.
+#[derive(Debug, Clone, Bpaf)]
+pub enum GenerationsCommands {
+    /// Prints help information
+    #[bpaf(command, hide)]
+    Help,
+
+    /// List generations of the selected environment
+    #[bpaf(command)]
+    List(#[bpaf(external(list::list))] List),
+}
+
+impl GenerationsCommands {
+    #[instrument(name = "generations", skip_all)]
+    pub fn handle(self, _config: Config, flox: Flox) -> Result<()> {
+        match self {
+            GenerationsCommands::Help => {
+                display_help(Some("generations".to_string()));
+            },
+            GenerationsCommands::List(args) => args.handle(flox)?,
+        }
+
+        Ok(())
+    }
+}
+
+fn try_get_generations_metadata(env: &ConcreteEnvironment) -> Result<AllGenerationsMetadata> {
```

Comment:
Suggestion: this might belong in `flox-rust-sdk` instead.

Reply:
addressed in [refactor: combine generations enabled env types under common enum](https://github.com/flox/flox/pull/3419/commits/315973e227157ee1fcf3d802112cfcd5a42016b4) and https://github.com/flox/flox/pull/3419#discussion_r2230945134